CityOfDetroit/bloom
# 78/add favorite listings capability

## Issue

- Addresses #78 

## Description

Creates three buttons, Sign in for Fav, Add to Favorites, Remove from Favorites on the front end.  Additionally it creates a favorites column in `user_preferences` on the back end.  Current iteration does not refresh the buttons once action has been taken.

## Type of change


## How Can This Be Tested/Reviewed?

Open a local instance of public website.  Inspect page so you can view console.  Visit the listings page while not signed in.  Notice the button asking you to sign in for favorites.
![image](https://user-images.githubusercontent.com/82653098/143274780-c63bc137-88b4-4be5-b39f-36f3d5afc0ee.png)

Next log into user account.  You will notice the listings page now has options for Adding and Removing favorites. The console log will show you the favorites array stored in the user_preference table.  Notice the buttons do not automatically update (bug) upon click.
![image](https://user-images.githubusercontent.com/82653098/143275078-6a097999-f292-4797-96f4-a49449886b2f.png)


## Checklist:
